hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/FieldOperations.java

```diff
@@ -31,831 +31,829 @@
+            @Override
+            public Object readObject(GenericRecord genericRecord, String fieldName) {
+                return ((InternalGenericRecord) genericRecord).getObjectArray(fieldName, Object.class);
+            }
 
-        @Override
-        public Object readIndexed(InternalGenericRecord record, String fieldName, int index) {
-            return record.getObjectFromArray(fieldName, index);
-        }
+            @Override
+            public Object readGenericRecordOrPrimitive(GenericRecord genericRecord, String fieldName) {
+                return genericRecord.getGenericRecordArray(fieldName);
+            }
 
-        @Override
-        public int hashCode(GenericRecord record, String fieldName) {
-            return Arrays.hashCode(record.getGenericRecordArray(fieldName));
-        }
+            @Override
+            public Object readIndexed(InternalGenericRecord record, String fieldName, int index) {
+                return record.getObjectFromArray(fieldName, index);
+            }
 
-        @Override
-        public void writeFieldFromRecordToWriter(DefaultCompactWriter writer, GenericRecord record, String fieldName) {
-            writer.writeGenericRecordArray(fieldName, record.getGenericRecordArray(fieldName));
-        }
+            @Override
+            public int hashCode(GenericRecord record, String fieldName) {
+                return Arrays.hashCode(record.getGenericRecordArray(fieldName));
+            }
 
-        @Override
-        public void writeJsonFormattedField(StringBuilder stringBuilder, AbstractGenericRecord record, String fieldName) {
-            GenericRecord[] objects = record.getGenericRecordArray(fieldName);
-            writeArrayJsonFormatted(stringBuilder, objects, StringBuilder::append);
-        }
-    });
+            @Override
+            public void writeFieldFromRecordToWriter(DefaultCompactWriter writer, GenericRecord record, String fieldName) {
+                writer.writeGenericRecordArray(fieldName, record.getGenericRecordArray(fieldName));
+            }
 
-    private static final FieldOperations[] ALL = FieldOperations.values();
+            @Override
+            public void writeJsonFormattedField(StringBuilder stringBuilder, AbstractGenericRecord record, String fieldName) {
+                GenericRecord[] objects = record.getGenericRecordArray(fieldName);
+                stringBuilder.append(Arrays.toString(objects));
+            }
+        };
+    }
 
-    private final FieldTypeBasedOperations operations;
+    private FieldOperations() {
 
-    FieldOperations(FieldTypeBasedOperations operations) {
-        this.operations = operations;
     }
 
-    public static FieldTypeBasedOperations fieldOperations(FieldType fieldType) {
-        FieldOperations fieldOperations = ALL[fieldType.getId()];
-        return fieldOperations.operations;
+    public static FieldKindBasedOperations fieldOperations(FieldKind fieldKind) {
+        return ALL[fieldKind.getId()];
```

Comment:
We are constructing the array with `.ordinal()` but getting with `.getId()`. I would prefer to use `.getId()` for both

Reply:
I prefer that as well. It was a leftover, I am changing it now.